dowhy/gcm/unit_change.py

```diff
@@ -0,0 +1,140 @@
+"""This module provides the APIs for attributing the change in the output value of a deterministic mechanism for a statistical unit.
+"""
+
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+import numpy as np
+import pandas as pd
+from sklearn.utils.validation import check_is_fitted
+from typing_extensions import Protocol
+
+from dowhy.gcm.shapley import ShapleyConfig, estimate_shapley_values
+
+
+class CanEvaluateInput(Protocol):
+    """This protocol defines a trait for classes that can evaluate a given input."""
+
+    @abstractmethod
+    def evaluate(self, input: np.ndarray) -> np.ndarray:
+        pass
```

Comment:
Wondering if we could make our interfaces/protocols more modular in the `fcms` module. There's the `InvertibleFunction` class which contains the same method.

Also, it seems that a mechanism here has a different interface than what `ProbabilisticCausalModel` currently uses for causal mechanisms. Not a huge problem, but it feels a bit inconsistent.

Not necessary to solve here, but I'm curious about your thoughts.

Reply:
Technically speaking, the mechanisms used here are deterministic ones, which are just the functions in usual additive noise models or more concretely `PredictionModel`. But there are subtle differences due to which I would not use `PredictionModel` for this API. 

- `Clone` interface of `PredictionModel` is not necessary for this use case. 
- I deliberately avoided the fitting part to keep the API lean as users may fit the prediction model on one data set, but ask for attribution on a different data set. Our APIs should be doing less of usual ML tasks like hyperparameter tuning, cross-validation, data pre-processing, etc to fit a function, where ever we do that now, we should find a way to detach them in the future if possible.

My recommendation would be to inherit the `CanEvaluateInput` protocol by both `InvertibleFunction` and `PredictionModel`.  `evaluate` method of `InvertibleFunction` and `predict` method of `PredictionModel` have the same signature, yet they just differ in semantics. Note that `predict` method is only used internally so renaming it to `evaluate` would not be a big issue.

Reply:
Revised the PR w/ existing interfaces.